### DISCUSSION TITLE
Warn if RedirectResponse is returned after awaiting client.connected

### DISCUSSION THREAD

**Discussion by RavioliSauce** [zauberzeug/nicegui#1743] [Ideas / Feature Requests]
### Question

Perhaps I'm using client.connected wrong, but it took me ages to figure out what was stopping my login page from redirecting to the home page. If you use the provided example, you should be able to reproduce it by going to the root page, it won't redirect to the /example page.
```python
from nicegui import ui, app, Client
from fastapi.responses import RedirectResponse
import asyncio

@ui.page('/')
async def index(client: Client) -> None:
    await client.connected()
    return RedirectResponse('/example')

@ui.page('/example')
async def example() -> None:
    ui.label('example page').classes('text-2xl')

ui.run(title='NiceGUI Example')
```
Like I said, I could be using the library wrong, if I am please let me know!

**Comment by rodja** [comment]
This is correct. If you await the client connection, NiceGUI will internally generate a html response which can then connect back via websocket to the server. Therefore the RedirectResponse can not be interpreted anymore. We should warn about this so you do not need to spend so much time on it to figure this out. I'll change the title and make this into a feature request.

**Reply by RavioliSauce** [reply]
Thank you so much for clarifying! Also, is there a way that I can still redirect after awaiting client.connected? I'm mostly doing this to store ip info!
edit:
Sorry if what I'm doing seem unusual or weird, I'm new to web dev and this library is incredibly powerful.

**Reply by rodja** [reply]
To open a new page after connection is established you can use https://nicegui.io/documentation/open. If you only need the IP address you could use https://github.com/zauberzeug/nicegui/issues/1695#issuecomment-1735981546 instead of getting it from the client.

**Reply by RavioliSauce** [reply]
I've been afraid to use ui.open because of the documentation saying all clients will open the url without the use of a button. But now that you've mentioned it I will try it out! Thank you!

**Reply by rodja** [reply]
Oh, the note does only apply for auto-index pages (eg. when you are not using a @page decorator). I've created https://github.com/zauberzeug/nicegui/pull/1760 to clarify this in the docs.

**Reply by RavioliSauce** [reply]
This makes sense! Thank you so much for your advice and help! And for making such a great library!
